### PR TITLE
Wip master

### PR DESCRIPTION
**Pull request by azat-co**
CC @AnnJohn @vicfriedman

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,43 @@
...
+In this lab, you will pass and access command-line arguments when working with Node.
+
+## Objectives
+
+1. Implement a Node script that takes CLI arguments and prints them back
+1. Use two arguments
+
+## Introduction
+
+Knowing how to pass and access command-line arguments is important in building CLI tools or small programs like the program which gives a random password.
```
Comment (victhevenot) [inline README.md]: Make the intro more project based - check out [this example](https://github.com/learn-co-curriculum/js-anonymous-functions-lab/blob/wp/README.md) for a refernce

File: README.md [hunk 2]
```diff
@@ -1 +1,43 @@
...
+
+![](password.png)
+
+## Instructions
+
+1. Implement a Node script `password.js` which takes CLI arguments and prints random passwords based on the arguments
+1. Use two arguments: length (first) and a number of passwords (second) to return so that `node password 2 2` and `node password 10 3` are valid commands
```
Comment (victhevenot) [inline README.md]: what do you mean number of passwords? `node 2 2` will give me two passwords, each with two letters?

File: README.md [hunk 3]
```diff
@@ -1 +1,43 @@
...
+2. Use `generatePassword()` function in the `password.js` to get random passwords, or write your own method
```
Comment (victhevenot) [inline README.md]: won't the test need to know what the function name is?

File: README.md [hunk 4]
```diff
@@ -1 +1,43 @@
...
+3. Finish program with `process.exit(0)`
+1. Run `$ npm test` to test
```
Comment (victhevenot) [inline README.md]: get rid of `$`

File: README.md [hunk 5]
```diff
@@ -1 +1,43 @@
...
+1. Implement a Node script `password.js` which takes CLI arguments and prints random passwords based on the arguments
+2. Use two arguments: length (first) and a number of passwords (second) to return so that `node password 2 2` and `node password 10 3` are valid commands
+3. Use `generatePassword()` function in the `password.js` to get random passwords, or write your own method
+4. Finish program with `process.exit(0)`
+5. Run `$ npm test` to test
+
+
+### Extra Info
+
+The password generator function:
```
Comment (victhevenot) [inline README.md]: don't give students this function - we want them to get as much practice coding as possible. they should be able to build this themselves at this point

**Comment by victhevenot** [comment]
comments - they're showing up as outdated from fixing typos, etc. but still relevant!

**Comment by victhevenot** [comment]
unclear where students are actually doing the coding here instead of just running code that is already written for them. we really want students to be writing as much code as possible to get practice (its the only way they'll learn and be developers) can we reframe some of these pre-written files with instructions for students to practice?

**Comment by vicfriedman** [comment]
hey @azat-co 

I'm still unclear here based on the instructions what work the student is actually supposed to be doing. 

We need to give students specific instructions about the nature of the code we want them to write. I noticed the solution console.logs an error if there isn't the correct amount of arguments, however that wasn't mentioned in the instructions. 

Your job is to process the data passed in by the two arguments, and pass the length of the passwords to the  `generatePassword` function, which generates the appropriate length password. the password should be comprised of all lower case letters.

Just remove the solution from the bottom of the Readme, students should be able to write that function. If they have issues, they have access to the solution branch to find it without just giving it to them in the readme.

Make sure wip-master get's this readme

**Comment by victhevenot** [comment]
hey @azat-co 

I went ahead and updated the intro - take a look for a good example at how to frame the work they're doing. I also went ahead and removed the `$`.

As for the student's work:

I'm still unclear here based on the instructions what work the student is actually supposed to be doing. 

We need to give students specific instructions about the nature of the code we want them to write. I noticed the solution console.logs an error if there isn't the correct amount of arguments, however that wasn't mentioned in the instructions. 

Something like this might be helpful:
"Your job is to process the data passed in by the two arguments, and pass the length of the passwords to the  `generatePassword` function, which generates the appropriate length password. the password should be comprised of all lower case letters." 

Just remove the solution from the bottom of the Readme, students should be able to write that function. If they have issues, they have access to the solution branch to find it without just giving it to them in the readme.

Make sure wip-solution get's this readme

And for the tests - I understand that we need a child process here to make sure that the commands work appropriately with command line arguments. Can we add an explanation to the lab readme here about what's going on with child process? Maybe make a comparison to running terminal commands in Ruby to base it off something they're already familiar with.

**Comment by azat-co** [comment]
updated CC @AnnJohn @vicfriedman

**Comment by victhevenot** [comment]
great intro and `child_process` explanation. 

A few things about child_process:
- can we add in what what arguments '140' and '1' are for? Those are the CLI arguments, we want 140 passwords of 1 character
- add a line about why we need child_process to begin with - so that we can test that the app takes in the command line arguments, etc.

Other than that - this is looking great! Make sure that wip-master gets this readme :)

Also, tests seem dependent on using `console.log` to print out the passwords, but the instructions say to "return" the passwords. make sure to have tests and instructions consistent

**Comment by azat-co** [comment]
@vicfriedman updated in both branches

**Pull request merged**